pkg/gitprovider/git_provider.go

```diff
@@ -128,6 +129,20 @@ func (a *AbstractGitProvider) parseSshGitUrl(gitURL string) (*StaticGitContext,
 	return repo, nil
 }
 
+// checkHTTPSAvailable checks if the given URL is accessible via HTTPS
+func checkHTTPSAvailable(source, owner, repo string) bool {
```

Comment:
I suggest that the full URL is passed as a parameter, not constructed in the function. This will reduce duplication and make `getCloneUrl` the only function that constructs the clone URL.

Reply:
@Tpuljak check the new commit, hope that works!

Reply:
@RS-labhub that's not what I meant. Only `checkHTTPSAvailable` should accept `url string`. `getCloneUrl` should still accept the same 3 arguments.

Also, please rename `checkHTTPSAvailable` to `isHttpsAvailable`.

Reply:
@Tpuljak could you check now? I hope that makes sense...